docs/docs/graphql.md

```diff
@@ -0,0 +1,158 @@
+
+```haskell
+{-# language ScopedTypeVariables, PartialTypeSignatures #-}
+
+libraryServer :: forall m. (MonadServer m)
+              => ServerT TypeMapping ServiceDefinition m _
+ = resolver
+      ( object @"Query"  ( method @"author" findAuthor
+                         , method @"books"  allBooks )
+      , object @"Author" ( field  @"id"     authorId
+                         , field  @"name"   authorName
+                         , field  @"books"  authorBooks )
+      , object @"Book"   ( field  @"id"     bookId
+                         , field  @"author" bookAuthor
+                         , field  @"title"  bookTitle ) )
+  where -- Query fields
+        findAuthor :: Text -> m (Maybe AuthorId)
+        allBooks   :: m [(BookId, AuthorId)]
+        -- Author fields
+        authorId    :: AuthorId -> m Int
+        authorName  :: AuthorId -> m Text
+        authorBooks :: AuthorId -> m [(BookId, AuthorId)]
+        -- Book fields
+        bookId     :: (BookId, AuthorId) -> m Int
+        bookAuthor :: (BookId, AuthorId) -> m AuthorId
+        bookAuthor (_, aid) = pure aid
+        bookTitle  :: (BookId, AuthorId) -> m Text
+        -- implementation
+```
+
+In the above code we have defined all fields in a big `where` block, but of course those may be defined as top-level functions, or inline in call to `field` or `method`.
+
+The final touch is to start the GraphQL server defined by `libraryServer`. The `Mu.GraphQL.Server` module defines tons of different ways to configure how the server behaves; the simplest option just requires a port and the name of the root type for queries.
+
+```haskell
+main = runGraphQLAppQuery 8080 libraryServer (Proxy @"Query")
+```
+
+## Subscriptions as streams
+
+## Comparison with other libraries
+
+There are other libraries targeting GraphQL server definition in Haskell.
+
+[`graphql-api`](https://github.com/haskell-graphql/graphql-api#readme) shares with Mu the encoding of the GraphQL schema in the type-level. In fact, as the [tutorial](https://haskell-graphql-api.readthedocs.io/en/latest/tutorial/Introduction.html) shows, its encoding is much closer to GraphQL's schema definition.
+
+```haskell
+type Hello
+  = Object "Hello" '[]
+           '[ Argument "who" Text :> Field "greeting" Text ]
+```
+
+This is expected: Mu's ability to target both RPC and GraphQL microservices means that sometimes there's some mismatch.
+
+[Morpheus GraphQL](https://morpheusgraphql.com/) also exposes GraphQL servers from Haskell code. Morpheus shared with Mu the ability to import a GraphQL schema into Haskell code. However, the types and fields are not represented by a type-level encoding, but *directly* as Haskell *records*.
```

Comment:
additionally, morpheus has GraphQL Client and Subscriptions, but i don't know how relevant is it for `mu-haskell' context.

ps. you don't have to define schema in morpheus. it will be generated by  Breadth-first search on haskell types.

Reply:
we have subscriptions too 😉